### ISSUE TITLE
Unhandled method textDocument/didOpen

### ISSUE THREAD

**Issue by baday19** [typescript-language-server/typescript-language-server#929]
I am trying to start the `server` and write the corresponding `client` in Python, but an unexpected error occurred while testing the didOpen notification. The error is as follows:

```json
{
  "jsonrpc": "2.0",
  "id": 2,
  "error": {
    "code": -32601,
    "message": "Unhandled method textDocument/didOpen"
  }
}
```

Here are my initialize request and didOpen request:

```python
# initialize
initialize_request = {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
        "processId": 3747,
        "rootUri": "file:///working/lstest",
        "capabilities": {
        }
    }
}
```

```python
# didOpen
open_request = {
    "jsonrpc": "2.0",
    "id": 2,
    "method": "textDocument/didOpen",
    "params": {
        "textDocument": {
            "uri": "file:///working/lstest/test.ts",
            "languageId": "typescript",
            "version": 1,
            "text": "function add(a: number, b: number): number {\n  return a + b;\n}\n\nadd(1, '2');"
        }
    }
}
```

Looking forward to receiving an answer, thank you very much!

**Comment by rchl** [comment]
Are you following the spec and waiting for the initialize request to respond before sending the didOpen request?

Please check the spec - https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize

**Comment by baday19** [comment]
> Are you following the spec and waiting for the initialize request response before sending the didOpen request?
> 
> Please check the spec - https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize



the response as follow:

```json
// Initialize Response
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "capabilities": {
      "textDocumentSync": 2,
      "completionProvider": {
        "triggerCharacters": [
          ".",
          "\"",
          "'",
          "/",
          "@",
          "<"
        ],
        "resolveProvider": true
      },
      "codeActionProvider": true,
      "codeLensProvider": {
        "resolveProvider": true
      },
      "definitionProvider": true,
      "documentFormattingProvider": true,
      "documentRangeFormattingProvider": true,
      "documentHighlightProvider": true,
      "documentSymbolProvider": true,
      "executeCommandProvider": {
        "commands": [
          "_typescript.applyWorkspaceEdit",
          "_typescript.applyCodeAction",
          "_typescript.applyRefactoring",
          "_typescript.configurePlugin",
          "_typescript.organizeImports",
          "_typescript.applyRenameFile",
          "_typescript.goToSourceDefinition"
        ]
      },
      "hoverProvider": true,
      "inlayHintProvider": true,
      "linkedEditingRangeProvider": false,
      "renameProvider": true,
      "referencesProvider": true,
      "selectionRangeProvider": true,
      "signatureHelpProvider": {
        "triggerCharacters": [
          "(",
          ",",
          "<"
        ],
        "retriggerCharacters": [
          ")"
        ]
      },
      "workspaceSymbolProvider": true,
      "implementationProvider": true,
      "typeDefinitionProvider": true,
      "foldingRangeProvider": true,
      "semanticTokensProvider": {
        "documentSelector": null,
        "legend": {
          "tokenTypes": [
            "class",
            "enum",
            "interface",
            "namespace",
            "typeParameter",
            "type",
            "parameter",
            "variable",
            "enumMember",
            "property",
            "function",
            "member"
          ],
          "tokenModifiers": [
            "declaration",
            "static",
            "async",
            "readonly",
            "defaultLibrary",
            "local"
          ]
        },
        "full": true,
        "range": true
      },
      "workspace": {
        "fileOperations": {
          "willRename": {
            "filters": [
              {
                "scheme": "file",
                "pattern": {
                  "glob": "**/*.{ts,js,jsx,tsx,mjs,mts,cjs,cts}",
                  "matches": "file"
                }
              },
              {
                "scheme": "file",
                "pattern": {
                  "glob": "**",
                  "matches": "folder"
                }
              }
            ]
          }
        }
      }
    }
  }
}
```

**Comment by rchl** [comment]
I'm asking if you are **waiting** for the response before sending the next one

**Comment by rchl** [comment]
There already exist lsp clients in Python. Maybe best to use something existing and working?

**Comment by baday19** [comment]
> the response before sending the next one

I'm sorry I didn't express myself clearly
I have already received the response before sending the next one

**Comment by baday19** [comment]
> There already exist lsp clients in Python. Maybe best to use something existing and working?

Okay, I'll give it a try. thanks!
